File: data/TiendaMusical/Dependencies/Producto.js
/**
 * Clase base que representa un producto en la tienda de música.
 */
class Producto {
    /**
     * @param {string} id 
     * @param {string} titulo 
     * @param {number} precio 
     */
    constructor(id, titulo, precio) {
        this.id = id;
        this.titulo = titulo;
        this.precio = precio;
    }

    /**
     * Calcula el precio con descuento.
     * @param {number} porcentaje Descuento a aplicar (debe estar entre 0 y 100)
     * @return {number} Precio con descuento
     * @throws {Error} Si el porcentaje no está entre 0 y 100
     */
    calcularDescuento(porcentaje) {
        if (porcentaje < 0 || porcentaje > 100) {
            throw new Error("El porcentaje debe estar entre 0 y 100");
        }
        return this.precio * (1 - porcentaje / 100);
    }

    // Getters y Setters
    getId() {
        return this.id;
    }
    
    getTitulo() {
        return this.titulo;
    }
    
    getPrecio() {
        return this.precio;
    }
    
    setPrecio(precio) {
        this.precio = precio;
    }
}

File: data/TiendaMusical/Producto.java
/**
 * Clase base que representa un producto en la tienda de música.
 */
public class Producto {
    private String id;
    private String titulo;
    private double precio;

    public Producto(String id, String titulo, double precio) {
        this.id = id;
        this.titulo = titulo;
        this.precio = precio;
    }

    /**
     * Calcula el precio con descuento.
     * @param porcentaje Descuento a aplicar
     * @return Precio con descuento
     */
    public double calcularDescuento(double porcentaje) {
        // TO DO
        return 0.0;
    }

    // Setters y Getters
    public String getId() { return id; }
    public String getTitulo() { return titulo; }
    public double getPrecio() { return precio; }
    public void setPrecio(double precio) { this.precio = precio; }
}